swig/matlab/helicsEndFederate.m

```diff
@@ -0,0 +1,19 @@
+function helicsEndFederate(fed_info)
+% helicsEndFederate helper function to abstract steps to close a HELICS federate
+%
+% TODO: mauy eventually become part of the C library and called via mex
+if isempty(fed_info)
+    warning('Empty fed_info, will not finalize or free HELICS federate. Will still close library')
+else
+    status = helics.helicsFederateFinalize(fed_info);
+    if status ~= 0
+        warning('Non-zero status (%d) returned in call to helicsFederateFinalize. Ignoring.', status)
+    end
+
+
+    helics.helicsFederateFree(fed_info);
+end
+
+%Note: helicsCloseLibrary is unrelated to the MATLAB library loading (from
+%helicsStartup)
+helics.helicsCloseLibrary();
```

Comment:
By including this call it will destroy all existing federates/cores/ brokers.  So if you have anything else from HELICS that you would want to keep this isn't going to work.

Reply:
Ok,  then lets add the parameter,  add some more details in the help results of the function stating when it is appropriate to use the function, and when it might not be appropriate, and the explanation of the argument.  

Also as part #427 I will be altering the closeLibrary function to go through and disconnect or terminate the federates, core, and brokers.